### PR TITLE
Change rocketchat::install API to allow setting the bundle URL directly

### PR DESCRIPTION
**Pull request by nmagedman**
I’m not clear on the difference between params.pp and install.pp (i.e. where should `$bundle_url` go?).
The code is not yet tested and I'm not strong in Puppet.

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -3,14 +3,23 @@
...
-    undef   => "https://rocket.chat/releases/${package_ensure}/download",
-    default => "${package_source}/rocket.chat-${package_ensure}.tgz",
+  # The API for generating the URL to fetch has gone through some evolution.
+  # To ease the transition, support the different formats:
+  # - v1: specify just $package_ensure (an oddly-named release identifier)
+  # - v2: specify $package_source (a base URL) and $package_ensure
+  # - v3: specify $bundle_url (the full, final URL)
+  $source = $bundle_url ? {
+    default => $bundle_url,
+    undef   => $package_source ? {
```
Comment (kapcod) [inline manifests/install.pp:19]: Please convert it to if/elsif/else one-level flow.

File: manifests/params.pp [hunk 2]
```diff
@@ -6,6 +6,7 @@
   $manage_repos       = true
   $download_path      = '/tmp'
   $destination        = '/opt'
+  $bundle_url         = undef
```
Comment (kapcod) [inline manifests/params.pp:9]: How is params.pp used?

**Review by kapcod — CHANGES_REQUESTED** [review]

File: manifests/params.pp (hunk 2 above)
Comment (kapcod) [reply manifests/params.pp:9]: Ok, I paid attention now to your message where you said "I’m not clear on the difference between params.pp".
@reist Do you know?

**Comment by nmagedman** [comment]
While I _do_ prefer being able to specify the URL _explicitly_ and _directly_ — rather than separately via a "source", an "ensure" (whatever _that_ is), and a buried, internally defined template — the _existing_ API is nonetheless sufficient for specifying the URL as we currently want it.

So this PR is not necessary, and I'm dropping it.

**Pull request closed without merging**